pkg/controllers/management/clusterupstreamrefresher/cluster_upstream_refresher.go

```diff
@@ -59,39 +59,24 @@ func (c *clusterRefreshController) onClusterChange(key string, cluster *mgmtv3.C
 		// not a KEv2 cluster
 		return cluster, nil
 	}
-
 	if !ready {
-		return cluster, nil
-	}
-
-	var lastRefreshTime string
-	var err error
-
-	if cluster.Annotations != nil {
-		lastRefreshTime = cluster.Annotations[clusterLastRefreshTime]
-	}
-
-	providerRefreshInterval, err := getProviderRefreshInterval(provider)
-	if err != nil {
-		return cluster, err
+		return c.updateCluster(cluster)
 	}
 
-	shouldRefresh, err := shouldRefreshCluster(providerRefreshInterval, lastRefreshTime)
+	providerRefreshInterval := getProviderRefreshInterval(provider)
+	shouldRefresh, err := shouldRefreshCluster(providerRefreshInterval, cluster.Annotations[clusterLastRefreshTime])
 	if err != nil {
 		return cluster, err
 	}
 
 	if shouldRefresh {
-		return cluster, err
-	}
-
-	cluster, err = c.refreshClusterUpstreamSpec(cluster, provider)
-	if err != nil {
-		return cluster, err
+		cluster, err = c.refreshClusterUpstreamSpec(cluster, provider)
+		if err != nil {
+			return cluster, err
+		}
 	}
 
 	c.clusterEnqueueAfter(key, providerRefreshInterval)
```

Comment:
If shouldRefresh is false should we enqueue after time.Now() - next-refresh-time? If a cluster should reenqueue every 5 minutes but is updates after 4m30s, then the next refresh could potential happen after 9m30s? Minimizing the window is good because the potential for races in this window is probably syncing's largest short coming. The docs also state that window is closely tied to the setting I believe.

Reply:
Yes, I agree. Updated.